File: problems/01-arrays-sequences/easy/lc-001-two-sum/solution.py
# LC-001 Two Sum

from typing import Sequence
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}

        for idx, num in enumerate(nums):
            complement = target - num
            if not complement in seen:
                seen[num] = idx
            else:
                return [idx, seen[complement]]

File: problems/01-arrays-sequences/easy/lc-001-two-sum/solution.cpp
/*
* LC-001 - Two Sum
*
*/

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int, int> seen;

        for (int i=0; i<nums.size(); i++) {
            int complement = target - nums[i];
            
            if (!seen.contains(complement)) {
                seen.insert({nums[i], i});
            }
            else {
                return std::vector {i, seen[complement]};
            }
        }
        return {};
    }
};
